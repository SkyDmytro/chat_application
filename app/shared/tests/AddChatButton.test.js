// import React from 'react';
// import renderer from 'react-test-renderer';
// import { AddChatButton } from '../AddChatButton';
// import { Pressable } from 'react-native';

// describe('AddChatButton', () => {
//   it('matches snapshot', () => {
//     const handlePress = jest.fn();
//     const tree = renderer.create(<AddChatButton onPress={handlePress} isDialogOpened={false} />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('calls onPress when pressed', () => {
//     const handlePress = jest.fn();
//     const component = renderer.create(<AddChatButton onPress={handlePress} isDialogOpened={false} />);
//     const instance = component.root;

//     renderer.act(() => {
//       instance.findByType(Pressable).props.onPress();
//     });

//     expect(handlePress).toHaveBeenCalled();
//   });
// });

import React from 'react';
import renderer from 'react-test-renderer';
import { AddChatButton } from '../AddChatButton';
import { Pressable, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

describe('AddChatButton', () => {
  it('matches snapshot when dialog is closed', () => {
    const handlePress = jest.fn();
    const tree = renderer.create(<AddChatButton onPress={handlePress} isDialogOpened={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when dialog is opened', () => {
    const handlePress = jest.fn();
    const tree = renderer.create(<AddChatButton onPress={handlePress} isDialogOpened={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const handlePress = jest.fn();
    const component = renderer.create(<AddChatButton onPress={handlePress} isDialogOpened={false} />);
    const instance = component.root;

    renderer.act(() => {
      instance.findByType(Pressable).props.onPress();
    });

    expect(handlePress).toHaveBeenCalled();
  });

});
