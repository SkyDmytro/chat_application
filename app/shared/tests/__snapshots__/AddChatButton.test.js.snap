// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddChatButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#007bff",
        "borderRadius": 50,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      false,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#fff",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "marginLeft": 5,
      }
    }
  >
    Add
  </Text>
</View>
`;

exports[`AddChatButton matches snapshot when dialog is closed 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#007bff",
        "borderRadius": 50,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      false,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#fff",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "marginLeft": 5,
      }
    }
  >
    Add
  </Text>
</View>
`;

exports[`AddChatButton matches snapshot when dialog is opened 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#007bff",
        "borderRadius": 50,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      false,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#fff",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "marginLeft": 5,
      }
    }
  >
    Close
  </Text>
</View>
`;
